app/src/amazonWebview/java/org/mozilla/focus/webview/FocusWebViewClient.java

```diff
@@ -111,9 +114,24 @@ public void onLoadResource(AmazonWebView view, String url) {
     // WebResourceRequest was added in API21 and there is no equivalent AmazonWebResourceRequest
     @Override
     public AmazonWebResourceResponse shouldInterceptRequest(AmazonWebView view, String request) {
+        if ((request != null && request.startsWith(APP_URL_PREFIX))) {
+            switch (request) {
+                case APP_URL_HOME:
+                    // Home screen should show a blank webview behind the overlay, but keep the url.
+                    callback.shouldInterceptRequest(request);
```

Comment:
nit: -> `onShouldInterceptRequest`. If you don't expect it to replace the method, you should make it clear this is a callback.